### ISSUE TITLE
CS1584, CS1658 from `inheritdoc cref` when referencing method with value tuple parameter

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#47268]
When `<inheritdoc cref="..."/>` refers to a member with a parameter whose type is a ValueTuple intrinsic, the completion list will help select the member while typing the cref attribute's value, but then C# will emit several warnings around the xml being invalid and the member not being found.

[Repro on SharpLab.io](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABATARgLACgUGYACdQgYUIG9dCbCB6BwgHgAcwAnMAW0IDtuApgF4ARC3YB7AObsBECCIB8TAJa8AFgPYrgsCQGNC+2QDNRAWTQAKAJIAFSTLkQKa4AF8AlCMJtg60To2Ti4AbRQUfi5hAHJxaVl5GIBdHzplII5uRWpaAmIAFkJzWwcE51VeYEVfR0SIT0pcmndmwjaGOmZg7j5BUXineSUAZQlo2vL5JkyQnJxaWpUANzBgAULi63s6ircawfrGqgXaVtOaDsZWLJ4o4TFd4eU1TW1dAyNTUQA5UqeXFY3NA3J4vD4/AERLNuOFIv04gCUmkMj0uPNFvkUEU/jsphBKtVJkMGk0LoRzosrl0biE+tEBgDRuMNocKjD0W1xCs1htsYQftsyiSmECqiCqp4DgDjm13FScO4gA==)

```cs
using System;
public class C {
    /// <param name="progress"><inheritdoc cref="M2(IProgress{int})" path="/param[@name='progress']" /></param>
    public void M(IProgress<int> progress) {
    }
    
    /// <param name="progress">Some progress</param>
    private void M2(IProgress<int> progress) {
    }
    
    /// <param name="progress"><inheritdoc cref="N(IProgress{(int,int)})" path="/param[@name='progress']" /></param>
    public void N(IProgress<int> progress) {
    }
    
    /// <param name="progress">Some progress</param>
    private void N2(IProgress<(int,int)> progress) {
    }    
}
```

**Expected**

no warnings

**Actual**

```
warning CS1584: XML comment has syntactically incorrect cref attribute 'N(IProgress{(int,int)})'
warning CS1658: Identifier expected. See also error CS1001.
warning CS1658: Syntax error, '>' expected. See also error CS1003.
warning CS1658: ) expected. See also error CS1026.
```

**Comment by AArnott** [comment]
Spelling out `ValueTuple<int,int>` solves the C# syntax and binding warnings, but it fails to actually inherit the xml doc for the named parameter.

**Comment by bencgreen** [comment]
I solved this by removing the argument types in the cref, e.g.

```csharp
/// <inheritdoc cref="QueryOptions.AddJoin{TFrom, TTo}(QueryParts, (TFrom table, Expression{Func{TFrom, string}} column), (TTo table, Expression{Func{TTo, string}} column), Func{QueryParts, IColumn, IColumn, QueryParts})"/>
```

to 

```csharp
/// <inheritdoc cref="QueryOptions.AddJoin"/>
```

Which has the advantage of being significantly easier to read - though doesn't work when there are multiple overrides.

**Comment by jonchardy** [comment]
The shortened syntax is nice, but doesn't work when the method is overloaded.

**Comment by zp-zpanda** [comment]
According to https://stackoverflow.com/questions/66448600/how-to-xml-document-tuple-list-ienumerablefoo-t-c-sharp:

To reference a method with value tuple parameters, replace the brackets (`(` and `)`) with `ValueTuple{` and `}`.

Example: `(int, string)` becomes `ValueTuple{int, string}`.

**Comment by AArnott** [comment]
Ya, that makes sense @zp-zpanda. Thanks for the workaround. This is still a bug though, at least for this part:

> the completion list will help select the member while typing the cref attribute's value

**Comment by zp-zpanda** [comment]
> Ya, that makes sense @zp-zpanda. Thanks for the workaround. This is still a bug though, at least for this part:
> 
> > the completion list will help select the member while typing the cref attribute's value
> 
> So the completion list itself is providing a faulty completion.

Yes, this is a possible workaround but docfx doesn't work with it for me.
